File: frontend/src/types/research_service.ts
export interface AiModel {
    id: number;
    model_key: string;
    default_params: Record<string, any>;
    max_tokens: number;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface ResearchServiceModel {
    id: number;
    service_id: number;
    model_id: number;
    is_default: boolean;
    created_at: string;
    updated_at: string;
    model: AiModel;
}

export interface ResearchService {
    id: number;
    service_key: string;
    name: string;
    description?: string | null;
    url?: string | null;
    default_model_id?: number | null;
    created_at: string;
    updated_at: string;
    default_model?: AiModel | null;
    service_models: ResearchServiceModel[];
}

export type ResearchServiceCreate = Pick<
    ResearchService,
    'service_key' | 'name' | 'description' | 'url' | 'default_model_id'
>; 

File: backend/app/schemas/research_service.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from app.schemas._base_model import CustomBaseModel

class ResearchServiceBase(CustomBaseModel):
    """Base schema for research services"""
    service_key: str
    name: str
    description: Optional[str] = None
    url: Optional[str] = None
    default_model_id: Optional[int] = None

class ResearchServiceCreate(ResearchServiceBase):
    """Schema for creating a new research service"""
    pass

class ResearchService(ResearchServiceBase):
    """Schema for a complete research service"""
    id: int
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    default_model: Optional["AiModelInResearchService"] = None
    # models: Optional[List["AiModelInResearchService"]] = None
    service_models: List["ResearchServiceModel"] = []

    model_config = ConfigDict(from_attributes=True)

# Avoid circular imports
from .ai_model import AiModelInResearchService
from .research_service_model import ResearchServiceModel 